pkg/armhelpers/util.go

```diff
@@ -39,19 +60,21 @@ func SplitBlobURI(URI string) (string, string, string, error) {
 
 // LinuxVMNameParts returns parts of Linux VM name e.g: k8s-agentpool1-11290731-0
 func LinuxVMNameParts(vmName string) (orchestrator string, poolIdentifier string, nameSuffix string, agentIndex int, err error) {
-	vmNameParts := strings.Split(vmName, "-")
-
-	if len(vmNameParts) != 4 {
+	vmNameParts := vmnameLinuxRegexp.FindStringSubmatch(vmName)
+	if len(vmNameParts) != 5 {
```

Comment:
This needs to remove. The point is it can be more than five.

Reply:
no, it cannot

Reply:
Discussed offline. This can remain.